src/main/java/com/walking/intensive/chapter1/task4/Task4.java

```diff
@@ -24,18 +24,41 @@
  */
 public class Task4 {
     public static void main(String[] args) {
-//        Для собственных проверок можете делать любые изменения в этом методе
         double a = 0;
-        double b = 0;
-        double c = 0;
-
+        double b = -2;
+        double c = 2;
         System.out.println(solveEquation(a, b, c));
-
     }
 
     static String solveEquation(double a, double b, double c) {
-        //        Место для вашего кода
+        if (a == 0 && b == 0 && c == 0) {
+            return "Бесконечное множество решений.";
+        } else if (a == 0 && b == 0) {
+            return "Количество решений: 0.";
+        } else if (a == 0) {
+            double root = -c / b;
+            return "Количество решений: 1. Корень: " + formatRoot(root);
+        }
+
+        double discriminantValue = b * b - 4 * a * c;
+
+        if (discriminantValue > 0) {
+            double root1 = (-b - Math.sqrt(discriminantValue)) / (2 * a);
+            double root2 = (-b + Math.sqrt(discriminantValue)) / (2 * a);
+            return "Количество решений: 2. Корни: " + formatRoot(Math.min(root1, root2)) + ";" + formatRoot(Math.max(root1, root2));
```

Comment:
Error: eckstyle] [ERROR] /home/runner/work/intensive-tasks-2024/intensive-tasks-2024/src/main/java/com/walking/intensive/chapter1/task4/Task4.java:48: Line is longer than 120 characters (found 132). [LineLength]